datasets/de/abgeordnetenwatch/crawler.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import json
+import math
+from zavod import Context
+from zavod import helpers as h
+
+
+def get_politician_data(context: Context, politician_api_url: str):
+    api_response = context.fetch_json(politician_api_url, cache_days=30)
+    politician = api_response.pop("data")
+    context.audit_data(politician)
+    return politician
+
+
+def crawl(context: Context):
+    # Fetch the source data URL specified in the metadata to a local path:
+    api_response = context.fetch_json(context.dataset.data.url, cache_days=30)
+    total_results = api_response.pop("meta").pop("result").pop("total")
+    num_batches = math.ceil(total_results / 100)
+
+    bi = 0
+    while bi < num_batches:
+        context.log.info(
+            "Get Politicians from range {} to {}".format(bi * 100, (bi + 1) * 100)
+        )
+
+        if bi > 0:
+            api_response = context.fetch_json(
+                context.dataset.data.url,
+                params={"range_start": bi * 100},
+                cache_days=30,
+            )
+            # We need to always recheck total number of results because the API
+            # returns a incorrect value in the first request without range_start
+            batch_total_results = api_response.pop("meta").pop("result").pop("total")
+            if batch_total_results != total_results:
+                num_batches = math.ceil(batch_total_results / 100)
+
+        for mandate in api_response.get("data"):
+            context.audit_data(mandate)
+            politician = mandate.pop("politician")
+
+            # Get state from parliament info
+            parliament_info = mandate.pop("parliament_period").pop("label").split(" ")
+            state = parliament_info[0]
+
+            # Create position
+            position = h.make_position(
+                context,
+                name="Member of Parliament",
```

Comment:
If I make name f"Member of the {state} Parliament", I get nice positions like this:

```json
    "positionOccupancies": [
      {
        "id": "de-aw-193679c9e97d9b0d555fa03321246b50828f5b87",
        "caption": "Occupancy",
        "schema": "Occupancy",
        "properties": {
          "post": [
            {
              "id": "de-aw-5270d360f089c5c49848b20c805f66d0caaf35ca",
              "caption": "Member of the Berlin Parliament",
              "schema": "Position",
              "properties": {
                "subnationalArea": [
                  "Berlin"
                ],
                "name": [
                  "Member of the Berlin Parliament"
                ],
                "country": [
                  "de"
                ]
              },
              "referents": [],
              "datasets": [
                "de_abgeordnetenwatch"
              ],
              "first_seen": "2023-08-25T11:39:04",
              "last_seen": "2023-08-25T11:39:04",
              "last_change": "2023-08-25T11:39:04",
              "target": false
            }
          ],
          "startDate": [
            "2023-05-03"
          ],
          "status": [
            "current"
          ],
          "holder": [
            "de-aw-0407b462289daa638b2be909d14ef07f4e4bb35a"
          ]
        },
```

Does this API contain information about the parliaments, so you can generate names like `Member of the Abgeordnetenhaus of Berlin` and `Member of the Landtag of Baden-Württemberg`?

Otherwise just including the state name like I did is fine.

Reply:
I couldn't find this extra information even in another endpoint, I changed to  "Member of the {state} Parliament"